github/resource_github_repository_ruleset.go

```diff
@@ -0,0 +1,550 @@
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+									"exclude": {
+										Type:        schema.TypeList,
+										Required:    true,
+										Description: "Array of ref names or patterns to exclude. The condition will not pass if any of these patterns match.",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"rules": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: "Rules within the ruleset.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"creation": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Only allow users with bypass permission to create matching refs.",
+						},
+						"update": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Only allow users with bypass permission to update matching refs.",
+						},
+						"deletion": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Only allow users with bypass permissions to delete matching refs.",
+						},
+						"required_linear_history": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Prevent merge commits from being pushed to matching branches.",
+						},
+						"required_deployments": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "Choose which environments must be successfully deployed to before branches can be merged into a branch that matches this rule.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+									"required_deployment_environments": {
+										Type:        schema.TypeList,
+										Required:    true,
+										Description: "The environments that must be successfully deployed to before branches can be merged.",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+						"required_signatures": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Commits pushed to matching branches must have verified signatures.",
+						},
+						"pull_request": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "Require all commits be made to a non-target branch and submitted via a pull request before they can be merged.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
```

Comment:
Is it because you put all the other values as `Optional`? Because all of `dismiss_stale_reviews_on_push`, `require_code_owner_review`, `require_last_push_approval`, `required_approving_review_count` and `required_review_thread_resolution` should be required

Reply:
Changed here: https://github.com/o-sama/terraform-provider-github/pull/2